### PR TITLE
Fix INTERNAL ASSERT FAILED when using weighted argmax local average

### PR DESCRIPTION
**Pull request by ben-hayes**
Addresses https://github.com/maxrmorrison/torchcrepe/issues/11

Simply uses the device of the incoming `logits` tensor to construct scalar tensors for comparison.

### CODE REVIEW

**Comment by maxrmorrison** [comment]
The fix + test works on my local machine. All set for merging. Thanks for the contribution!
